# First Project - Profitable App Profiles for the App Store & Google Play Markets

The project goal is to analyze data to help our developers understand what type of apps are likey to attract more users.

This is a portfolio project and can be shared to prospective employers.

In [1]:
apple_list = []
google_list = []

def open_file_for_analysis(file_name):
    opened_file = open(file_name, encoding='utf8')
    from csv import reader
    read_file = reader(opened_file)
    apps_data = list(read_file)
    
    return apps_data

apple_list = open_file_for_analysis('AppleStore.csv')
google_list = open_file_for_analysis('googleplaystore.csv')

## Print a few rows for the two csv files

In [2]:
print(apple_list[0:3])

[['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic'], ['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'], ['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']]


In [3]:
print(google_list[0:3])

[['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'], ['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'], ['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']]


In [4]:
print("Google List Count")
len(google_list)

Google List Count


10842

In [5]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))


In [6]:
explore_data(apple_list,0,3,rows_and_columns=True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 7198
Number of columns: 16


# Apple List Notes:
The apple list have 7,198 rows as of 05/03/2020. The file also has 16 columns. The data definition of the file are all described in https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps

In [7]:
print('\n')
explore_data(google_list,0,3,rows_and_columns=True)



['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


Number of rows: 10842
Number of columns: 13


# Google List Notes:
The google list has 10,842 rows as of 05/03/2020. The file also has 13 columns. The data definition of the file are all described in https://www.kaggle.com/lava18/google-play-store-apps#googleplaystore.csv

Data Cleaning Task for file "googleapplestore.csv". 
Per the discussion, users have found a missing column. The current row for my copy of the csv file has the error on row 10,473.

The lesson's instruction is to delete the row. However, after inspecting the data, the values for the row is clean but is only missing a column. My approach would be not to delete the row but to move the data in their respective columns.

Before deleting or modifying rows, it would be good to check what other rows thay may have a missing column assuming that a column without a value should have a null value.

Additional steps that I would perform outside of python, would be to import data into a database table, and use that to further look into the data to validate the source using two different approaches. In the database, I would use SQL statements to determine duplicate rows for all columns in a row.

In [8]:
header_len = len(google_list[0])
print("Header lenght is " + str(header_len))
for row in google_list[1:]:
    if len(row) < header_len:
        print("Lenght of row is " + str(len(row)))
        print("Index of row is " + str(google_list.index(row)))
        print(row)

Header lenght is 13


# Converting the above loop to a function that accepts a list as a parameter

In [9]:
def check_invalid_row(list):
    header_len = len(list[0])
    print("Header lenght is " + str(header_len))
    for row in list[1:]:
        if len(row) < header_len:
            print("Lenght of row is " + str(len(row)))
            print("Index of row is " + str(list.index(row)))
            print(row)


# Check invalid rows for the googleplaystore.csv file. Expected result is that no rows returned since the file was fixed.

In [10]:
check_invalid_row(apple_list)

Header lenght is 16


# Check for duplicate app names in the google list

To separate duplicate rows, the duplicate is going to be assigned to a new list.


In [11]:
duplicate_apps = []
unique_apps = []
applist_list = []
google_list = []

def open_file_for_analysis(file_name):
    opened_file = open(file_name, encoding='utf8')
    from csv import reader
    read_file = reader(opened_file)
    apps_data = list(read_file)
    
    return apps_data

def check_duplicate_app(app_list):
    for row in app_list[1:]:
        name = row[0]
        if name in unique_apps:
            duplicate_apps.append(name)
        else:
            unique_apps.append(name)
    
    print('Number of duplicate apps:', len(duplicate_apps))
    print('\n')
    print('Duplicate app names:', duplicate_apps[:5]) 
    print('\n')    
    return(unique_apps)
    

google_list = open_file_for_analysis('googleplaystore.csv')
google_unique_list = check_duplicate_app(google_list)

print('Number of google unique apps:', len(google_unique_list))
print('\n')
print('Unique google app names:', google_unique_list[:5]) 
print('\n')


Number of duplicate apps: 1181


Duplicate app names: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings']


Number of google unique apps: 9660


Unique google app names: ['Photo Editor & Candy Camera & Grid & ScrapBook', 'Coloring book moana', 'U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'Sketch - Draw & Paint', 'Pixel Draw - Number Art Coloring Book']




In [12]:
for app in google_list:
    name = app[0]
    if name == 'Box':
        print(app)
      

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


# Using panda DataFrame to check all the columns for duplicate rows

In [13]:
import pandas as pd

print('Columns for google_list')
print(google_list[0])
print('\n')
print('Google List Count')
print(len(google_list))

dfObj1 = pd.DataFrame(google_list)

print('\n')
print('Display duplicate rows using DataFrame')
print('\n')
duplicateRows = dfObj1[dfObj1.duplicated(keep='last')]
print('Count of duplicate rows in the Dataframe')
print(len(duplicateRows))
print('\n')
print(duplicateRows)


Columns for google_list
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


Google List Count
10842


Display duplicate rows using DataFrame


Count of duplicate rows in the Dataframe
483


                                  0                    1    2       3  \
165                    Ebook Reader  BOOKS_AND_REFERENCE  4.1   85842   
193   Docs To Go™ Free Office Suite             BUSINESS  4.1  217730   
194              Google My Business             BUSINESS  4.4   70991   
205                             Box             BUSINESS  4.2  159872   
214             ZOOM Cloud Meetings             BUSINESS  4.4   31614   
...                             ...                  ...  ...     ...   
3788     ABC News - US & World News   NEWS_AND_MAGAZINES  4.0   18976   
3789                       NBC News   NEWS_AND_MAGAZINES  4.1   63020   
3791                      USA TODAY   NEWS_AND_MAGA

# Print some rows for the duplicates
The list only displays the last duplicate rows. There are 3 rows for app equals to 'Box. With panda, only the 2 duplicate rows are contained in the list.

In [14]:
duplicate_only = []
duplicate_only = duplicateRows.values.tolist()

print(len(duplicate_only))

for row in duplicate_only:
    name = row[0]
    if name == 'Box':
        print(row)
        print('\n')

483
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']




# Removing duplicate rows using panda

After removing the duplicates, data frame was converted to a list. I am retaining an original copy of the google play store in another list. In this approach, I am considering the duplicate row by comparing each column.

In [15]:
dfObj2 = dfObj1.drop_duplicates()
print(dfObj2)

clean_google_list = []
clean_google_list = dfObj2.values.tolist()

print('\n')
print('New google list count')
print(len(clean_google_list))
print('\n')

for row in clean_google_list[:5]:
    print(row)
    print('\n')


                                                       0                    1  \
0                                                    App             Category   
1         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
2                                    Coloring book moana       ART_AND_DESIGN   
3      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
4                                  Sketch - Draw & Paint       ART_AND_DESIGN   
...                                                  ...                  ...   
10837                                   Sya9a Maroc - FR               FAMILY   
10838                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10839                             Parkinson Exercices FR              MEDICAL   
10840                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10841      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

            2        3     

# Expected Result

The expected result from the lessons does not match the counts based on my analysis. For example, the previous task that splits the duplicate and unique rows only counts the app names. Out of the 1,181 rows there are actually 483 duplicate rows only. For example there are 3 rows for the app equal to 'Box'.

With the panda duplicate rows, the duplicates are removed from the google list and leaves the unique row.

Using the panda drop_duplicates removes all the duplicate rows. Below is the total

In [16]:
print('Length of google play store list after cleanup')
print(len(clean_google_list[1:]))

Length of google play store list after cleanup
10358


In [17]:
print(len(google_list))

10842


# Removing duplicate rows with the use of a dictionary and only keeping the row with the max review count

Create a dictionary and only keep rows with the max review count. In my google (android list), my expected result is 9660 instead of 9,659. I didn't remove the row at the beginning of the excercise that was missing a column. I fixed the row by adding a cell instead of removing it. If the row was deleted it would match the expected count of 9,659.

In [18]:
reviews_max = {}
clean_google_list2 = []

for row in google_list[1:]:
    name = row[0]
    n_reviews = float(row[3])
    if name not in reviews_max:
        reviews_max[name] = n_reviews
    elif reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    
print("Number of Rows without the duplicates")
print(len(reviews_max))
print('\n')

for row in google_list[1:]:
    name = row[0]
    n_reviews = float(row[3])
    
    for y in reviews_max:
        if name == y and reviews_max[y] == n_reviews:
            clean_google_list2.append(row)
            break
        elif name == y and name not in reviews_max:
            clean_google_list2.append(row)   

print("Number of rows after cleanup")
print(len(clean_google_list2))

Number of Rows without the duplicates
9660


Number of rows after cleanup
10055


## Based on the task for this mission, I have validated the results above with a SQL query using a temp table where I have stored the google_play_list. Returned count from the query is also 10,055.

SQL used to check in Postgres:

    select *
    from google_play_store a,
    (
        select app, max(reviews) reviews
        from
        (
        select distinct app, reviews
        from google_play_store
        --
        --
        order by app
        ) y
        --
        --where app like 'A&E%'
        --
        group by app ) b
        --
    where a.app = b.app
    and a.reviews = b.reviews;

## The count from this task only accounts for the max(reviews) count if more than one row for an app exist.

### The following task filters the rows from the cleaned google app list. The goal is to only keep english apps from the list. A new list will be created just for the english apps.

In [19]:
google_english_app = []
google_nonenglish_app = []
apple_english_app = []
apple_nonenglish_app = []
google_english_app2 = []
google_nonenglish_app2 = []

def check_char(app_name):
    allowed_over_127 = 0
    for x in range(len(app_name)):
        if ord(app_name[x]) <= 127:
            english_app = True
        else:
            if allowed_over_127 < 3:
                allowed_over_127 += 1
                english_app = True
            else:
                english_app = False
                
    
    return english_app

def populate_new_list(app_list, index):
    english_app = []
    non_english_app = []
    for row in app_list:
        english_only = check_char(row[index])
        if english_only:
            english_app.append(row)
        else:
            non_english_app.append(row)
    
    return english_app, non_english_app
       

apple_english_app, apple_nonenglish_app = populate_new_list(apple_list, 1)
google_english_app, google_nonenglish_app = populate_new_list(clean_google_list2, 0)
    

print("Number of Google english apps")
print(len(google_english_app))
print("")
print("Number of Google non-english apps")
print(len(google_nonenglish_app))

print("\n")
print("List of Google non-english apps")
print("\n")
for row in google_nonenglish_app[:5]:
    print(row[0])

print("\n")
print("Number of Apple english apps")
print(len(apple_english_app))
print("")
print("Number of Apple non-english apps")
print(len(apple_nonenglish_app))

print("\n")
print("List of Apple non-english apps")
print("\n")
for row in apple_nonenglish_app[:5]:
    print(row[1])


Number of Google english apps
10028

Number of Google non-english apps
27


List of Google non-english apps


Flame - درب عقلك يوميا
РИА Новости
RMEduS - 음성인식을 활용한 R 프로그래밍 실습 시스템
AJ렌터카 법인 카셰어링
中国語 AQリスニング


Number of Apple english apps
6357

Number of Apple non-english apps
841


List of Apple non-english apps


爱奇艺PPS -《欢乐颂2》电视剧热播
聚力视频HD-人民的名义,跨界歌王全网热播
优酷视频
网易新闻 - 精选好内容，算出你的兴趣
淘宝 - 随时随地，想淘就淘


### This task will separate the free apps from the paid apps using the list for the english apps.

In [20]:
google_english_free_apps = []
google_english_paid_apps = []

apple_english_free_apps = []
apple_english_paid_apps = []

for row in google_english_app:
    if row[7] == '0':
        google_english_free_apps.append(row)
    else:
        google_english_paid_apps.append(row)
        
for row in apple_english_app:
    if row[4] == '0.0':
        apple_english_free_apps.append(row)
    else:
        apple_english_paid_apps.append(row)
    
print("Number of Free English Google Apps")
print(len(google_english_free_apps))
print("\n")
print("Number of Paid English Google Apps")
print(len(google_english_paid_apps))
print("\n")
print("Number of Free English Apple Apps")
print(len(apple_english_free_apps))
print("\n")
print("Number of Paid English Apple Apps")
print(len(apple_english_paid_apps))

Number of Free English Google Apps
9243


Number of Paid English Google Apps
785


Number of Free English Apple Apps
3357


Number of Paid English Apple Apps
3000


## Profitable apps

In this task we need to identify apps that are profitable for the Apple Store and Google Play. The following criteria will be used for the apps:

1) Google Play

Columns for this data set that can be used for the analysis is the category and genres. Doing a count for each prime_genre frequency can be used.

2) Apple Playstore

Columns for this data set that can be used for the analysis is the prime_genre.


### Printing the frequency table with frequencies expressed as percentages.

The new freq_table function will require two arguments: dataset and index. This function should return the result for the frequency.

Frequency tables:

1) App Store - Prime genre column

2) Google Play Store - Genres and Category columns.



In [21]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
        
        
def freq_table(dataset, index):
    content_ratings = {}
    total = 0

    for row in dataset:
        total += 1
        content_key = row[index]
        if content_key in content_ratings:
            content_ratings[content_key] += 1
        else:
            content_ratings[content_key] = 1
            
            
    content_percentage = {}
    for key in content_ratings:
        percentage = (content_ratings[key] / total) * 100
        content_percentage[key] = percentage
    
    return content_percentage

In [22]:
print("\n")
print("Example of rows from English Apple App store")

for row in apple_english_free_apps[:3]:
    print("\n")
    print(row)

print("\n")
print("Apple English App Frequency for Genres ")

display_table(apple_english_free_apps, 11)




Example of rows from English Apple App store


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Apple English App Frequency for Genres 
Games : 58.117366696455164
Entertainment : 7.953529937444147
Photo & Video : 4.885314268692285
Education : 3.544831694965743
Social Networking : 3.276735180220435
Shopping : 2.5915996425379806
Utilities : 2.5320226392612453
Sports : 2.0554066130473636
Music : 1.966041108132261
Health & Fitness : 1.966041108132261
Productivity : 1.7277330950253202
Lifestyle : 1.668156091748585
News : 1.3404825737265416
Travel : 1.2213285671730711
Finance : 1.1915400655347037
Food & Drink : 0.923

In [23]:
print("\n")
print("Example of rows from Google App store")

for row in google_english_free_apps[:3]:
    print("\n")
    print(row)




Example of rows from Google App store


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


### Findings for the Apple Store Frequency:

For the App store, the 3 top genres are for Games (58%), Entertainment (8% rounded off) , and Photo & Video (5% rounded off).

The education genre didn't make it to the top for the free apps in the Apple store. The catalog genre has the lowest and is less than 1%. The percentages for the popularity of the apps does not represent the largest number of users.


In [24]:
print("\n")
print("Google English App Frequency for Column Genre")

display_table(google_english_free_apps, 9)



Google English App Frequency for Column Genre
Tools : 8.114248620577733
Entertainment : 6.112733960835227
Education : 5.355404089581304
Business : 4.727902196256626
Productivity : 3.884020339716542
Lifestyle : 3.851563345234231
Medical : 3.7109163691442175
Finance : 3.591907389375744
Sports : 3.5486313967326626
Health & Fitness : 3.2889754408741747
Communication : 3.2456994482310937
Personalization : 3.213242453748783
Action : 3.0076814886941468
Photography : 2.8994915070864438
News & Magazines : 2.8886725089256737
Social : 2.693930542031808
Shopping : 2.5100075732987124
Travel & Local : 2.41263658985178
Dating : 2.120523639510981
Books & Reference : 2.0880666450286705
Simulation : 1.969057665260197
Arcade : 1.7851346965271015
Video Players & Editors : 1.6985827112409388
Casual : 1.6985827112409388
Maps & Navigation : 1.3523747700962891
Food & Drink : 1.2225467921670454
Puzzle : 1.1143568105593422
Racing : 0.9520718381477875
Strategy : 0.897976847343936
Role Playing : 0.8979768473439

In [25]:
print("\n")
print("Google English App Frequency for Column Category")

display_table(google_english_free_apps, 1)



Google English App Frequency for Column Category
FAMILY : 18.457210862274152
GAME : 9.455804392513253
TOOLS : 8.125067618738505
BUSINESS : 4.727902196256626
PRODUCTIVITY : 3.884020339716542
LIFESTYLE : 3.8623823433950015
MEDICAL : 3.7109163691442175
FINANCE : 3.591907389375744
SPORTS : 3.47289840960727
HEALTH_AND_FITNESS : 3.2889754408741747
COMMUNICATION : 3.2456994482310937
PERSONALIZATION : 3.213242453748783
PHOTOGRAPHY : 2.8994915070864438
NEWS_AND_MAGAZINES : 2.8886725089256737
SOCIAL : 2.693930542031808
SHOPPING : 2.5100075732987124
TRAVEL_AND_LOCAL : 2.4234555880125503
DATING : 2.120523639510981
BOOKS_AND_REFERENCE : 2.0880666450286705
VIDEO_PLAYERS : 1.7202207075624798
MAPS_AND_NAVIGATION : 1.3523747700962891
EDUCATION : 1.3199177756139782
FOOD_AND_DRINK : 1.2225467921670454
ENTERTAINMENT : 1.179270799523964
LIBRARIES_AND_DEMO : 0.897976847343936
AUTO_AND_VEHICLES : 0.8871578491831656
HOUSE_AND_HOME : 0.8763388510223954
WEATHER : 0.7681488694146922
EVENTS : 0.6815968841285297

### Findings for the Google Store Frequency:

The genres for entertainment are more popular than the apps designed for the practical purposes. The genres for entertainment are more popular than the apps designed for the practical purposes.

The app store has more apps that are for families.


### The following task is to create a frequency for the number of users to find out the apps with most users.

Apple Free Store Frequency for Users

- Prime Genre (11)
- Rating count total (5)

In [26]:
apple_genre_list = freq_table(apple_english_free_apps, 11 )

for genre in apple_genre_list:
    total = 0
    len_genre = 0

    for app in apple_english_free_apps:
        app_row = app[11]
        rating = app[5]
        
        if app_row == genre:
            app_rating = float(rating)
            total += app_rating
            len_genre += 1

    average = total / len_genre
    print(genre, ' : ', average)    


Social Networking  :  68951.51818181819
Photo & Video  :  27747.84756097561
Games  :  21889.392619169656
Music  :  57326.530303030304
Reference  :  70997.78947368421
Health & Fitness  :  22945.015151515152
Weather  :  52279.892857142855
Utilities  :  17819.44705882353
Travel  :  27558.048780487807
Shopping  :  25992.597701149425
News  :  20303.666666666668
Navigation  :  57482.11111111111
Lifestyle  :  15013.92857142857
Entertainment  :  13349.794007490636
Food & Drink  :  27958.709677419356
Sports  :  23008.898550724636
Book  :  34788.6875
Finance  :  28321.325
Education  :  6945.126050420168
Productivity  :  20318.275862068964
Business  :  7074.944444444444
Catalogs  :  3203.2
Medical  :  525.4285714285714


In [27]:
print("\n")
print("Apple app with the highest users")
for row in apple_english_free_apps:
    if row[11] == 'Reference':
        print(row[1], ' : ', row[5])



Apple app with the highest users
Bible  :  985920
Dictionary.com Dictionary & Thesaurus  :  200047
Dictionary.com Dictionary & Thesaurus for iPad  :  54175
Google Translate  :  26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran  :  18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition  :  17588
Merriam-Webster Dictionary  :  16849
Night Sky  :  12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE)  :  8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools  :  4693
GUNS MODS for Minecraft PC Edition - Mods Tools  :  1497
Guides for Pokémon GO - Pokemon GO News and Cheats  :  826
WWDC  :  762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free  :  718
VPN Express  :  14
Real Bike Traffic Rider Virtual Reality Glasses  :  8
教えて!goo  :  0
Jishokun-Japanese English Dictionary & Translator  :  0
無料で音楽や写真・カメラの裏技アプリ for iPhone7  :  0


The result for the Google English Free apps above shows that Reference has the highest number of ratings.

Google Play Store Frequency for Users evaluated based on the following columns:

- Category (1)
- Installs (5)

In [28]:
google_genre_list = freq_table(google_english_free_apps, 1)

for category in google_genre_list:
    total = 0
    len_category = 0
    total_app_rating = 0

    for app_row in google_english_free_apps:
        category_row = app_row[1]
        rating = app_row[5]
        
        if category_row == category:

            app_rating = rating.replace(',','')
            app_rating = app_rating.replace('+','')
            total_app_rating = float(app_rating)
            total += total_app_rating
            len_category += 1

    average = total / len_category
    print(category, ', ', average)


ART_AND_DESIGN ,  1952105.1724137932
AUTO_AND_VEHICLES ,  647317.8170731707
BEAUTY ,  513151.88679245283
BOOKS_AND_REFERENCE ,  8657695.647668393
BUSINESS ,  1887876.6361556065
COMICS ,  817657.2727272727
COMMUNICATION ,  52706387.336666666
DATING ,  854027.8418367347
EDUCATION ,  2265163.9344262294
ENTERTAINMENT ,  11105137.614678899
EVENTS ,  253542.22222222222
FINANCE ,  1374888.9518072288
FOOD_AND_DRINK ,  2015387.1769911505
HEALTH_AND_FITNESS ,  4261507.90131579
HOUSE_AND_HOME ,  1737067.4197530865
LIBRARIES_AND_DEMO ,  638503.734939759
LIFESTYLE ,  1415698.6806722688
GAME ,  17784747.654462244
FAMILY ,  3752940.615474795
MEDICAL ,  141184.85131195336
SOCIAL ,  24252457.437751003
SHOPPING ,  8119993.900862069
PHOTOGRAPHY ,  17952495.57835821
SPORTS ,  3715983.436137072
TRAVEL_AND_LOCAL ,  14204036.098214285
TOOLS ,  10800324.199733688
PERSONALIZATION ,  5990693.225589226
PRODUCTIVITY ,  20319301.710306406
PARENTING ,  542603.6206896552
WEATHER ,  5074486.197183099
VIDEO_PLAYERS , 

Based on the results for the Google English Free apps, the category with the highest installs are apps within the COMMUNICATION category.

In [29]:
print("\n")
print("Google app with the highest users")
for row in google_english_free_apps:
    if row[1] == 'COMMUNICATION':
        print(row[0], ' : ', row[5])



Google app with the highest users
WhatsApp Messenger  :  1,000,000,000+
Messenger for SMS  :  10,000,000+
My Tele2  :  5,000,000+
imo beta free calls and text  :  100,000,000+
Contacts  :  50,000,000+
Call Free – Free Call  :  5,000,000+
Web Browser & Explorer  :  5,000,000+
Browser 4G  :  10,000,000+
MegaFon Dashboard  :  10,000,000+
ZenUI Dialer & Contacts  :  10,000,000+
Cricket Visual Voicemail  :  10,000,000+
TracFone My Account  :  1,000,000+
Xperia Link™  :  10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard  :  10,000,000+
Skype Lite - Free Video Call & Chat  :  5,000,000+
My magenta  :  1,000,000+
Android Messages  :  100,000,000+
Google Duo - High Quality Video Calls  :  500,000,000+
Seznam.cz  :  1,000,000+
Antillean Gold Telegram (original version)  :  100,000+
AT&T Visual Voicemail  :  10,000,000+
GMX Mail  :  10,000,000+
Omlet Chat  :  10,000,000+
My Vodacom SA  :  5,000,000+
Microsoft Edge  :  5,000,000+
WhatsApp Messenger  :  1,000,000,000+
Messenger – Text 